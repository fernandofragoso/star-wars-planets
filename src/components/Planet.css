
.planet {
  transition: opacity 0.5s;
  position: relative;
  text-shadow: 2px 2px 2px #333;
}

.planet header{
  text-align: center;
  text-transform: uppercase;
}

.planet article{
  padding: 30px 0;
  font-size: 20px;
  line-height: 30px;
  text-align: left;
}

.planet article .center{
  margin-top: 10px;
  text-align: center;
}

.planet--loading {
  opacity: 0;
}

.planet--loading:after {
  content: '';
  position: absolute;
  bottom: 0;
  right: 0;
}