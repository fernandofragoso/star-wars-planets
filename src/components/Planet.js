import React, { Component } from 'react';
import './Planet.css';

export default class Planet extends Component {
  render() {
    let planetClasses = (this.props.loading ? "planet planet--loading" : "planet");
    const { population, climate, terrain, films } = this.props;

    return (
      <div className={planetClasses}>
        <header>
          <h1>{this.props.name}</h1>
        </header>
        <article>
          <div><strong>Population: </strong>{this._formatNumber(population)}</div>
          <div><strong>Climate: </strong>{climate}</div>
          <div><strong>Terrain: </strong>{terrain}</div>
          <div className="center">{this._formatFilms(films)}</div>
        </article>
      </div>
    );
  }

  //Format numbers for better visualization
  _formatNumber(number) {
    if (number !== "unknown") {
      return number.toString().split("").reverse().map((char, index) => {
        return ((index%3===0) ? `${char}.` : char);
      }).reverse().join("").slice(0, -1);
    }
    return number
  }

  //Format films information
  _formatFilms(films) {
    switch (parseInt(films)) {
      case 0:
        return "Never featured in a film";
      case 1:
        return `Featured in ${films} film`;
      default:
        return `Featured in ${films} films`;
    }
  }
}