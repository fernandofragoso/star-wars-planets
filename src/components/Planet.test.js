import React from 'react';
import ReactDOM from 'react-dom';
import Planet from './Planet';
import renderer from 'react-test-renderer';

let getPlanet = (population = "10000", films = "1") => {
  return <Planet
    name="name"
    population={population}
    climate="climate"
    terrain="terrain"
    films={films}
    loading="false"
    error="false" />
}

let mockPlanet = <Planet
  name="name"
  population="10000"
  climate="climate"
  terrain="terrain"
  films="1"
  loading="false"
  error="false" />

it('renders correctly', () => {
  const tree = renderer.create(getPlanet()).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without population and films', () => {
  const tree = renderer.create(getPlanet("unknown", 0)).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with multiple films', () => {
  const tree = renderer.create(getPlanet("9000000000", 10)).toJSON();
  expect(tree).toMatchSnapshot();
});