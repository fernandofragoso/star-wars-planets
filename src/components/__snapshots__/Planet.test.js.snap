// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="planet planet--loading"
>
  <header>
    <h1>
      name
    </h1>
  </header>
  <article>
    <div>
      <strong>
        Population: 
      </strong>
      10.000
    </div>
    <div>
      <strong>
        Climate: 
      </strong>
      climate
    </div>
    <div>
      <strong>
        Terrain: 
      </strong>
      terrain
    </div>
    <div
      className="center"
    >
      Featured in 1 film
    </div>
  </article>
</div>
`;

exports[`renders correctly with multiple films 1`] = `
<div
  className="planet planet--loading"
>
  <header>
    <h1>
      name
    </h1>
  </header>
  <article>
    <div>
      <strong>
        Population: 
      </strong>
      9.000.000.000
    </div>
    <div>
      <strong>
        Climate: 
      </strong>
      climate
    </div>
    <div>
      <strong>
        Terrain: 
      </strong>
      terrain
    </div>
    <div
      className="center"
    >
      Featured in 10 films
    </div>
  </article>
</div>
`;

exports[`renders correctly without population and films 1`] = `
<div
  className="planet planet--loading"
>
  <header>
    <h1>
      name
    </h1>
  </header>
  <article>
    <div>
      <strong>
        Population: 
      </strong>
      unknown
    </div>
    <div>
      <strong>
        Climate: 
      </strong>
      climate
    </div>
    <div>
      <strong>
        Terrain: 
      </strong>
      terrain
    </div>
    <div
      className="center"
    >
      Never featured in a film
    </div>
  </article>
</div>
`;
